dvc/project.py

```diff
@@ -3,8 +3,10 @@
 import dvc.prompt as prompt
 import dvc.logger as logger
 
-from dvc.exceptions import DvcException, MoveNotDataSourceError
-from dvc.exceptions import NotDvcProjectError
+from dvc.stage import Stage
```

Comment:
Almost every method uses `stage`, @shcheklein, that's why I took it outside;

- add
- checkout
- destroy
- fetch
- import
- lock
- move
- pipeline
- pull
- remove
- repro
- run
- status
- unlock

Also, for `help` to be shown, it doesn't need to load `dvc/project` because it should halt the execution during the `parseArgs`;

By the way, I tried to profile `dvc --help` with [cprofilev](https://github.com/ymichael/cprofilev), I've never done profiling in Python before, so I really don't know how to read the [results](https://gist.github.com/mroutis/1c78a5da7a3acca40499c06fb5aa18bb).

Although, seing `azure` and `boto3` on the list caught my attention.

Reply:
@mroutis Could you please check that `dvc`(no args) and `dvc --help` take the same time with and without this change? I'm afraid that always loading `stage.py` might affect the startup time for python.

Reply:
@efiop , here the results:

**RFC**:
```
.venv ❯ for i in {1..10}; do time dvc &> /dev/null; done
dvc &> /dev/null  0.60s user 0.08s system 99% cpu 0.684 total
dvc &> /dev/null  0.63s user 0.07s system 99% cpu 0.701 total
dvc &> /dev/null  0.65s user 0.06s system 99% cpu 0.711 total
dvc &> /dev/null  0.65s user 0.08s system 99% cpu 0.725 total
dvc &> /dev/null  0.60s user 0.05s system 99% cpu 0.642 total
dvc &> /dev/null  0.60s user 0.06s system 99% cpu 0.661 total
dvc &> /dev/null  0.62s user 0.05s system 99% cpu 0.674 total
dvc &> /dev/null  0.58s user 0.08s system 100% cpu 0.658 total
dvc &> /dev/null  0.60s user 0.05s system 99% cpu 0.648 total
dvc &> /dev/null  0.57s user 0.07s system 99% cpu 0.643 total
```

**master**:
```
.venv ❯ for i in {1..10}; do time dvc &> /dev/null; done
dvc &> /dev/null  0.56s user 0.06s system 100% cpu 0.611 total
dvc &> /dev/null  0.55s user 0.05s system 99% cpu 0.608 total
dvc &> /dev/null  0.55s user 0.05s system 99% cpu 0.602 total
dvc &> /dev/null  0.56s user 0.05s system 99% cpu 0.613 total
dvc &> /dev/null  0.55s user 0.06s system 100% cpu 0.614 total
dvc &> /dev/null  0.53s user 0.07s system 100% cpu 0.608 total
dvc &> /dev/null  0.55s user 0.06s system 100% cpu 0.611 total
dvc &> /dev/null  0.57s user 0.03s system 99% cpu 0.606 total
dvc &> /dev/null  0.55s user 0.06s system 99% cpu 0.612 total
dvc &> /dev/null  0.54s user 0.07s system 99% cpu 0.611 total
```

It's quite significant, indeed; forgot about `CmdBase` and `CmdConfig` which requires to load a `Project` to use the locking mechanism.

I'll remove the respective changes!